### DISCUSSION TITLE
Portweiterleitung an VLAN an zweiter Fritzbox  - Storj

### DISCUSSION THREAD

**Discussion by dancekid** [Freetz-NG/freetz-ng#634] [Allgemein]
Hallo liebe Community,
zum Betrieb von 3 getrennten Storj-Nodes benötigt man 3 externe IPs + 3 unterschiedliche Subnetze.

Mein Plan:
Node A physisch: bei mir -> extern erreichbar: sohn.dyn.de Port (TCP+UDP)-> Port 10 -> VLAN1 -> 192.168.177.10 Port 10
Node B physisch: bei mir -> extern erreichbar: mutter.dyn.de (TCP+UDP) -> Port 11 -> VLAN2 -> 192.168.178.10 Port 10
Node C physisch: bei mir -> extern erreichbar: vater.dyn.de (TCP+UDP) -> Port 12 -> VLAN3 -> 92.168.179.10 Port 10
Mutter = FB4040, Sohn = FB4040, Vater = FB7490

Demnach brauche ich in der FB4040 Mutter eine Portweiterleitung an sohn.dyn.de oder per Wireguard an VLAN2
Meine FB4040 muss dann entweder je Port von extern an ein anderes VLAN weiterleiten oder die interne Wireguardverbindung
muss zwischen VLAN2 <-> Mutter und VLAN3 <-> Vater aufgebaut werden.

dazu meine Fragen:
Ist das mit Freetz machbar ? Welche Pakete brauche ich dazu ?

Einmal die reine Weiterleitung FB4040+FB7490 und einmal die VLAN Aufteilung + die Weiterleitung an das entsprechende VLAN anhand der ankommenden Ports mit einer FB4040.
Bei der Variante mit Wireguard brauche ich auf der FB4040 die VLAN Aufteilung + 2 Wireguard Verbindungen mit VLAN2+3 Endpunkten.

Grüße in den Abend
Michael

**Comment by fda77** [comment]
Also bevor ich da was um die komischen AVM-Daemons herum basteln müsste, würde ich zu Openwrt greifen, falls es kompatibel ist. Die 4040 scheint sogar unterstützt, 7490 nur "halb". Damit kannst du Vlan (und vieles mehr) wie bei jedem normalen Linux Gerät konfigurieren

**Comment by dancekid** [comment]
Guten Morgen fda77,
das kann ich für die Box hier bei mir vor Ort auch machen. Notfalls auch bei der Box von meiner Mutter.
Mein Vater nutzt die 7490 allerdings mit Telefonie und braucht daher die AVM Funktionalitäten.

Habe mir gestern alle Packetbeschreibungen von A-Z durchgelesen. Geht die Weiterleitung eventuell mit BIRD ?

Ich brauche ja nur in der 7490:
ist:  Port 10 ankommend -> Weiterleitung an interne IP Port 10 (im Standard auf lokales Netz begrenzt).
soll: Port 10 ankommend -> Weiterleitung an sohn.dyn.de Port 10

Gruß Michael

**Comment by fda77** [comment]
Ja, mit openwrt geht das ganze dect/isdn/pots verloren.
Ich bin mir jetzt aber nicht sicher ob du überhaupt VLAN für das brauchst was du vor hast! Ein Weiterleitung auf IP-eben kann man nicht über den Hostnamen machen. Daher sind wechselnde IPs ungünstig. (aber es sind ja schonbar interne statische).
Wie kommt die Fritzbox überhaupt in die Vlan 1-3 rein?

**Comment by dancekid** [comment]
Hallo fda77,
wenn eine Weiterleitung an eine dyndns adresse nicht funktioniert :-(

Mir fällt dann nur die Variante mit Wireguard ein.
Meine Box dann mit OpenWRT und 3 VLANs. 192.168.167.1 + 192.168.168.1 + 192.168.169.1
Die 7490 meines Vaters mit Wireguard zu VLAN2
Die 4040 meiner Mutter mit Wireguard zu VLAN3

Dann brauche ich ein Freetz Package für die Eltern Boxen mit Weiterleitung UPD + TCP über Wireguardtunnel auf eine IP in meinen VLAN2 + VLAN3.
Bei meiner Mutter kann ich die 4040 mit OpenWRT flashen; damit sollte es auch funktionieren. Telefon machen die über das Kabelmodem.

Die 7490 meines Vaters halt nicht, da DECT der Box genutzt wird.

https://bird.network.cz/?get_doc&v=20&f=bird-1.html#ss1.1
Bird ist ja ein Paket für Freetz. So wie ich das verstehe, kann man damit eingehende Datenpakete weiterrouten.
Nur habe ich dort keine Eingrenzung auf einzelne TCP + UDP Ports gefunden.

Gruß Michael

**Comment by fda77** [comment]
Ich versteh nicht was das werden soll. Du brauchst du gar keine vlans. Wenn du eh einen "hauptrouter" hast und dann für jeden je einen "unterrouter" sehen sich die Geräte doch nicht.
Vlans würden nur die Unterrouter ersparen (hab ich so, er Switch verteile die Vlans des Routers)

**Comment by dancekid** [comment]
Hallo fda,

die Hardware (3 storj-nodes) soll bei mir stehen und ich den Strom zahlen.
Die Einstiegspunkte ins Internet je Node 1x bei mir, 1x bei meiner Mutter, 1x bei meinem Vater.
Die Nodes prüfen, ob sie a) in unterschiedlichen Subnetzen sind und b) ob die öffentliche IP Adresse unterschiedlich ist.
Somit muss ich 3 interne VLANs haben und 3 externe IP-Adressen.
Die Nodes lassen dyndns-Adressen zu und brauchen je 1x TCP 1x UDP Port.

Zur Zeit habe ich noch meinen alten DSL Anschluss und den neuen Glasfaseranschluss
damit habe ich 2x externe IP per dyndns und 2x interne Subnetze (192.168.178.1 + 192.168.188.1)
Die Nodes sind komplett getrennt. Der Traffic wird einzeln nach Geschwindigkeit berechnet (max 100 Mbit pro Node).
Wenn ich beide Nodes an meinem Glasfaseranschluss mit 500 Mbit betreibe, dann habe ich 100 Mbit für beide Nodes (50 Mbit pro).
Das Programm rechnet pro Subnet alles als 1 Node.
Also muss ich die Hardware entsprechend in unterschiedliche private IP-Ranges stellen und extern an unterschiedlichen Punkten ins Internet lassen.
Eine Möglichkeit ist den Traffic von meiner Mutter für die benötigten Ports an mich weiterzuleiten und ihre externe IP zu nutzen.

**Comment by fda77** [comment]
Ich würde nichts auf der Fritzbox machen. Einfach ein Gerät in deinem Lan (zb PI) das zu den 2 externen VPN aufbaut und auf diese 2 Client VPN-IPs die Portforwards bekommt. Auch ein Forwared vom lokalen Anschluss. So hat man alle Ports auf 1 Gerät und könnte diese zb an isoliete Container/VMs auf diesem Gerät weiterleiten. Oder in VLans packen wenn es wirklich verschieden Geräte sein sollen

**Comment by dancekid** [comment]
Guten Morgen fda,
die normale SW auf der 7490 kann doch die Ports nicht weiterleiten an eine IP außerhalb des lokalen Netzes. 
Sonst hätte ich ja schon FB-Wireguard Vater zu OpenWRT VLAN2 Sohn probiert.
In der Oberfläche der Fritzbox kann ich immer nur ein Gerät für die Portweiterleitung 192.168.178.xx auswählen.
Nichtmal Gastnetz funktioniert.

**Reply by fda77** [reply]
> SW auf der 7490 kann doch die Ports nicht weiterleiten an eine IP außerhalb des lokalen Netze

Hab das nie ausprobiert. Gilt das auch für VPN-Clients? Und wie sieht es mit VPN-Clients aus wenn man Wireguard oder Openwrt von Freetz als Server nimmt?
Vielleicht brauchst du iptables (bekanntlich geht mit AVM Software der conntrack/state Bereich nicht)

**Comment by aus-sua** [comment]
Guten Tag,

ich lese hier nur mit und sorry, mir erscheint das alles sehr "verkopft", wie Du die
Aufteilung der drei LAN machen möchtest.
Aber sei es drum, wenn Du damit zurecht kommst und es "händeln" kannst...

Ich würde es zumindest "sortenrein" lösen:
kaufe Dir noch 'ne dritte 4040, bespiele alle mit OpenWRT und hänge die 7490
(nur als DECT-"Zuspieler", IP-Client) dahinter.

Ist Dir das zu teuer, dann:
DECT können ja auch genug ältere Boxen:
7312, 7412, 7360v2, 7362SL ...
die gibt es gebraucht (für ca. so 10,-), dann OpenWRT auf die 7490, die alte
Box als VoIP/DECT dahinter und gut ist es...

Je nach Firmwarestand sind die DECT-Funktionen dieser Boxen etwas
"beschnitten bzw. hinterher", sollte Dein Vater wirklich diese neuen Funktionen
benötigen, dann eben OpenWRT auf die ältere Box (7360v2, 7362SL o.ä.) und
die 7490 wieder nur als DECT-"Zuspieler" dahinter.
Ist nicht so 100%-tig sortenrein, aber für 10,-...
;-)

**Reply by fda77** [reply]
Die 7490 ist für Openwrt noch nicht ganz Endusertauglich. Sie hat 2 CPUs und die 2. macht nur Wlan, man braucht also "2x Openwrt". Seit Jahren gibt es da diverse PRs, zb: https://github.com/openwrt/openwrt/pull/5075
